s3/src/main/java/ch/cyberduck/core/s3/S3Session.java

```diff
@@ -200,14 +197,18 @@ protected RequestEntityRestStorageService connect(final Proxy proxy, final HostK
             configuration.setServiceUnavailableRetryStrategy(new S3TokenExpiredResponseInterceptor(this,
                     new ThreadLocalHostnameDelegatingTrustManager(trust, host.getHostname()), key, prompt));
         }
+        // TODO check with DK: what's the criterion on host to go for AssumeRoleWithWebIdentity?
```

Comment:
It should be possible to make the decision based on some property in the connection profile.

Reply:
@dkocher what would be a good naming convention for the following items and which should we make configurable:
* `stsEndpointURL` -> `.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(String.format("https://sts.amazonaws.com"), null))`
* `stsTTL`-> `.withDurationSeconds`
*  `stsRole`-> `.withRoleArn("arn:aws:iam::930717317329:role/google-Test-Role")`
* `stsRoleSessionName` -> `.withRoleSessionName("cyberduck-test");`

Reply:
All lowercase properties with prefix `s3.sts.`.